src/coreComponents/mesh/generators/InternalMeshGenerator.cpp

```diff
@@ -543,18 +543,32 @@ static void getElemToNodesRelationInBox( ElementType const elementType,
  * @param partition
  * @param domain
  */
-void InternalMeshGenerator::generateMesh( DomainPartition & domain )
+MeshGeneratorHelper InternalMeshGenerator::generateCellBlockManager( CellBlockManager & cellBlockManager )
 {
   GEOSX_MARK_FUNCTION;
+  MeshGeneratorHelper meshGeneratorHelper;
+  meshGeneratorHelper.setHasMetisNeighborList( false );
+  SpatialPartition partition = meshGeneratorHelper.getSpatialPartition();
```

Comment:
Instead of using `SpatialPartition` directly here, what about simplifying `MeshGeneratorHelper` by simply providing `min` and `max`? It looks like this is the only information used.

Reply:
Later on we use other part of the spatialpartition :
```
grep --color -nH --null -e "\bpartition\b" InternalMeshGenerator.cpp
InternalMeshGenerator.cpp:543: * @param partition
InternalMeshGenerator.cpp:551:  SpatialPartition partition = partitionDescriptor.getSpatialPartition();
InternalMeshGenerator.cpp:563:    partition.setSizes( m_min, m_max );
InternalMeshGenerator.cpp:615:  // Get the first and last indices in this partition each direction
InternalMeshGenerator.cpp:625:      if( partition.isCoordInPartition( elemCenterCoords[i][k], i ) )
InternalMeshGenerator.cpp:636:        if( partition.isCoordInPartition( elemCenterCoords[i][k], i ) )
InternalMeshGenerator.cpp:644:  // Calculate number of elements in this partition from each region, and the
InternalMeshGenerator.cpp:730:  reduceNumNodesForPeriodicBoundary( partition, numNodesInDir );
InternalMeshGenerator.cpp:757:          setNodeGlobalIndicesOnPeriodicBoundary( partition, globalIJK );
InternalMeshGenerator.cpp:821:    if( std::any_of( partition.m_Periodic.begin(), partition.m_Periodic.end(), []( int & dimPeriodic ) { return dimPeriodic == 1; } ) )
InternalMeshGenerator.cpp:904:                // Fix local connectivity for single theta (y) partition (radial meshes only)
InternalMeshGenerator.cpp:985:  // 2) first local element in component partition is zero
```

Reply:
Later on we use the SpatialPartition more:
```
grep --color -nH --null -e "\bpartition\b" InternalMeshGenerator.cpp
InternalMeshGenerator.cpp:543: * @param partition
InternalMeshGenerator.cpp:551:  SpatialPartition partition = partitionDescriptor.getSpatialPartition();
InternalMeshGenerator.cpp:563:    partition.setSizes( m_min, m_max );
InternalMeshGenerator.cpp:615:  // Get the first and last indices in this partition each direction
InternalMeshGenerator.cpp:625:      if( partition.isCoordInPartition( elemCenterCoords[i][k], i ) )
InternalMeshGenerator.cpp:636:        if( partition.isCoordInPartition( elemCenterCoords[i][k], i ) )
InternalMeshGenerator.cpp:644:  // Calculate number of elements in this partition from each region, and the
InternalMeshGenerator.cpp:730:  reduceNumNodesForPeriodicBoundary( partition, numNodesInDir );
InternalMeshGenerator.cpp:757:          setNodeGlobalIndicesOnPeriodicBoundary( partition, globalIJK );
InternalMeshGenerator.cpp:821:    if( std::any_of( partition.m_Periodic.begin(), partition.m_Periodic.end(), []( int & dimPeriodic ) { return dimPeriodic == 1; } ) )
InternalMeshGenerator.cpp:904:                // Fix local connectivity for single theta (y) partition (radial meshes only)
InternalMeshGenerator.cpp:985:  // 2) first local element in component partition is zero
```